migrate/migrate.go

```diff
@@ -148,12 +152,44 @@ func runScripts(pool *sql.DB, scripts map[string]string, ignoreFiles []string) e
 		filenames = append(filenames, name)
 	}
 	sort.Strings(filenames)
+
 	for _, file := range filenames {
+		content, ok := scripts[file]
+		if !ok {
+			continue
+		}
+
+		var currentHash string
+		if err := pool.QueryRow("SELECT hash FROM migration_logs WHERE path = $1", file).Scan(&currentHash); err != nil && !errors.Is(err, sql.ErrNoRows) {
+			return err
+		}
+
+		hash := sha1.Sum([]byte(content))
+		if string(hash[:]) == currentHash {
+			logger.Tracef("Skipping script %s", file)
+			continue
+		}
+
 		logger.Tracef("Running script %s", file)
 		if _, err := pool.Exec(scripts[file]); err != nil {
 			return fmt.Errorf("failed to run script %s: %w", file, err)
 		}
+
+		if _, err := pool.Exec("INSERT INTO migration_logs(path, hash) VALUES($1, $2) ON CONFLICT (path) DO UPDATE SET hash = $2", file, hash[:]); err != nil {
+			return fmt.Errorf("failed to save migration log %s: %w", file, err)
+		}
 	}
 
 	return nil
 }
+
+func createMigrationLogTable(pool *sql.DB) error {
+	query := `CREATE TABLE IF NOT EXISTS migration_logs (
+		path VARCHAR(255) NOT NULL,
+		hash bytea NOT NULL,
+		updated_at TIMESTAMPTZ DEFAULT NOW() NOT NULL,
+		PRIMARY KEY (path)
+	)`
+	_, err := pool.Exec(query)
+	return err
+}
```

Comment:
Why aren't we using the tables/*.hcl way for this ?

Reply:
Atlas schemas are run after running the `functions` scripts. And this table needs to be there even before we run any scripts.

Reply:
can't we do `functions` -> `schema` -> then scripts

Reply:
That's how it is right now. We keep track of migration of `functions` as well